src/api/streams/updateStream.js

```diff
@@ -2,7 +2,11 @@
 import type { ApiResponse, Auth } from '../apiTypes';
 import { apiPatch } from '../apiFetch';
 
-export default (auth: Auth, id: number, property: string, value: string): Promise<ApiResponse> =>
-  apiPatch(auth, `streams/${id}`, res => res, {
-    [property]: value,
-  });
+export type ParamsType = {
+  new_name?: string,
+  description?: string,
+  is_private?: boolean,
+};
+
+export default (auth: Auth, id: number, newValues: ParamsType): Promise<ApiResponse> =>
+  apiPatch(auth, `streams/${id}`, res => res, { ...newValues });
```

Comment:
Do you need to spread the values?
```{ ...newValues }`` vs just `newValues`

Reply:
I tried with `newValues `, then flow is giving error.
````
Cannot call apiPatch with newValues bound to params because:
 • string [1] is incompatible with boolean [2] in property description.
 • string [1] is incompatible with number [3] in property description.
 • boolean [4] is incompatible with string [5] in property is_private.
 • boolean [4] is incompatible with number [3] in property is_private.
 • string [6] is incompatible with boolean [2] in property new_name.
 • string [6] is incompatible with number [3] in property new_name.

           src/api/streams/updateStream.js
       [6]  6│   new_name?: string,
       [1]  7│   description?: string,
       [4]  8│   is_private?: boolean,
            9│ };
           10│
           11│ export default (auth: Auth, id: number, newValues: ParamsType): Promise<ApiResponse> =>
           12│   apiPatch(auth, `streams/${id}`, res => res, newValues);
           13│

           src/utils/url.js
 [5][2][3] 16│ export type UrlParams = { [string]: string | boolean | number };



Found 6 errors
```


In `toggleMobilePushSettings` also we are creating new object.
```
export default async ({
  auth,
  opp,
  value,
}: {
  auth: Auth,
  opp: string,
  value: boolean,
}): Promise<ApiResponse> =>
  apiPatch(auth, 'settings/notifications', res => res, {
    ...getRequestBody(opp, value),
  });
```
So I thought of doing same 😅